src/workerd/io/actor-cache.c++

```diff
@@ -2325,56 +2321,78 @@ kj::Promise<void> ActorCache::startFlushTransaction() {
     batch.wordCount += words;
   };
 
-  kj::Vector<CountedDeleteFlush> countedDeleteFlushes;
+  kj::Vector<CountedDeleteFlush> countedDeleteFlushes(countedDeletes.size());
+  for (auto countedDelete : countedDeletes) {
+    if (countedDelete->isFinished) {
+      // This countedDelete has already be executed, but we haven't delivered the final count to
+      // the waiter yet. We'll skip it here since the destructor of CountedDeleteWaiter should
+      // eventually remove this entry from `countedDeletes`.
+      continue;
+    }
+
+    auto& countedDeleteFlush = countedDeleteFlushes.add(CountedDeleteFlush{
+      .countedDelete = kj::addRef(*countedDelete),
+    });
+    // We might have successfully deleted these entries, but had the broader transaction fail.
+    // In that case, we might have entries that have since been overwritten, and which no longer
+    // need to be scheduled for deletion.
+    kj::Vector<kj::Own<Entry>> entriesToDelete(countedDelete->entries.size());
+    for (auto& entry : countedDelete->entries) {
+      if (entry->overwritingCountedDelete && countedDelete->completedInTransaction) {
```

Comment:
Given this check, is it possible for a `CountedDeleteFlush` to end up with zero entries in `entriesToDelete`? If so, will anything bad happen when we later go to use it?

Might be better to delay the `countedDeleteFlushes.add()` call from above until after we've confirmed `entriesToDelete` is non-empty?

Also, if we do that, a test case covering it would be nice, assuming it doesn't require anything terribly convoluted.